### ISSUE TITLE
crane agent error in aws eks

### ISSUE THREAD

**Issue by wswz963852** [gocrane/crane#540]
**Describe the bug**
container logs:
[agent.log](https://github.com/gocrane/crane/files/9467589/agent.log)

```yaml
  - containerID: docker://0b930d6ab9e27271594f16efbf937d4c8d8921a3d9ef6a360d3950e54a43b077
    image: gocrane/crane-agent:v0.7.0
    imageID: docker-pullable://gocrane/crane-agent@sha256:6041ca97172d3216918b481f9ec17f584430b5687c5bb170b5fd632ac0b02f94
    lastState:
      terminated:
        containerID: docker://e3fd4d924adbef8881023bdfca36736de52924680982dae4ea14a58311f00430
        exitCode: 2
        finishedAt: "2022-09-01T04:52:12Z"
        reason: Error
        startedAt: "2022-09-01T03:59:52Z"
    name: crane-agent
    ready: true
    restartCount: 25
    started: true
    state:
      running:
        startedAt: "2022-09-01T04:52:13Z"
```

**Reproduce steps**

helm repo add crane https://gocrane.github.io/helm-charts
helm repo update
helm install crane -n crane-system --create-namespace crane/crane -f crane.yaml

crane.yaml:
```yaml
craned:
  image:
    repository: docker.io/gocrane/craned
    pullPolicy: IfNotPresent
    # Overrides the image tag whose default is the chart appVersion.
    tag: ""

  replicaCount: 1

  containerArgs:
    prometheus-address: http://rancher-monitoring-prometheus.cattle-monitoring-system.svc:9090
    feature-gates: Analysis=true,TimeSeriesPrediction=true,Autoscaling=true
    v: 2

  podAnnotations: { }
  resources: { }
  nodeSelector: { }
  tolerations: [ ]
  affinity: { }

metricAdapter:
  image:
    repository: docker.io/gocrane/metric-adapter
    pullPolicy: IfNotPresent
    # Overrides the image tag whose default is the chart appVersion.
    tag: ""
  replicaCount: 1
  containerArgs:
    secure-port: 6443
    alsologtostderr: true
    v: 2
    api-qps: 300
    api-burst: 400
  remoteAdapterEnabled: true
  remoteAdapterArgs:
    remote-adapter-service-namespace: cattle-monitoring-system
    remote-adapter-service-name: rancher-monitoring-prometheus-adapter
    remote-adapter-service-port: 443

  podAnnotations: { }
  resources: { }
  nodeSelector: { }
  tolerations: [ ]
  affinity: { }

craneAgent:
  enable: true
  image:
    repository: docker.io/gocrane/crane-agent
    pullPolicy: IfNotPresent
    # Overrides the image tag whose default is the chart appVersion.
    tag: ""
  podAnnotations: { }
  resources: { }
  nodeSelector: { }
  tolerations:
  - operator: Exists
  affinity: { }
  containerArgs:
    v: 2

cranedDashboard:
  enable: true
  image:
    repository: docker.io/gocrane/dashboard
    pullPolicy: IfNotPresent
```

**Environment (please complete the following information):**
 - K8S Version: 1.21
 - Crane Version: 0.7.0

**Comment by chenkaiyue** [comment]
You can build the agent image with the latest code, concurrent map read and map write err has been fixed few days ago.

**Comment by qmhu** [comment]
fixed in v0.8.0.
